tyk-hybrid/templates/deployment-pump.yaml

```diff
@@ -23,106 +23,100 @@ spec:
-      - name: TYK_PMP_PUMPS_HYBRID_META_USESSL
-        value: "{{ .Values.gateway.rpc.useSSL }}"
-      - name: TYK_PMP_PUMPS_HYBRID_META_SSLINSECURESKIPVERIFY
-        value: "{{ .Values.gateway.rpc.sslInsecureSkipVerify }}"
-      - name: TYK_PMP_PUMPS_HYBRID_META_GROUPID
-        value: "{{ .Values.gateway.rpc.groupId }}"
-
-      {{- if .Values.pump.extraEnvs }}
-      {{- toYaml .Values.pump.extraEnvs| nindent 10 }}
-      {{- end }}
-    command: ["/opt/tyk-pump/tyk-pump", "-c", "/etc/tyk-pump/pump.conf"]
-    volumeMounts:
-      - name: tyk-pump-conf
-        mountPath: /etc/tyk-pump
-      {{- if .Values.pump.mounts }}
-      {{- range $secret := .Values.pump.mounts }}
-      - name: {{ $.Release.Name }}-pump-secret-{{ $secret.name }}
-        mountPath: {{ $secret.mountPath }}
-      {{- end }}
-      {{- end }}
-    resources:
-  {{ toYaml .Values.pump.resources | indent 12 }}
-securityContext:
-  runAsUser: 1000
-  fsGroup: 2000
-volumes:
-  - name: tyk-pump-conf
-    configMap:
-      name: pump-conf-{{ include "tyk-hybrid.fullname" . }}
-      items:
-        - key: pump.conf
-          path: pump.conf
-  {{- if .Values.pump.mounts }}
-  {{- range $secret := .Values.pump.mounts }}
-  - name: {{ $.Release.Name }}-pump-secret-{{ $secret.name }}
-    secret:
-      secretName: {{ $.Release.Name }}-pump-secret-{{ $secret.name }}
-  {{- end }}
-  {{- end }}
-  {{- end}}
+{{- if .Values.pump.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.pump.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.pump.tolerations }}
+      tolerations:
+{{ toYaml .Values.pump.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.pump.affinity }}
+      affinity:
+{{ toYaml .Values.pump.affinity | indent 8 }}
+{{- end }}
+      containers:
+      - name: pump-{{ .Chart.Name }}
+        image: "{{ .Values.pump.image.repository }}:{{ .Values.pump.image.tag }}"
+        imagePullPolicy: {{ .Values.pump.image.pullPolicy }}
+        workingDir: "/opt/tyk-pump"
+        securityContext:
+          runAsNonRoot: true
+          allowPrivilegeEscalation: false
+          privileged: false
+          readOnlyRootFilesystem: true
+          capabilities:
+            drop:
+              - all
+        env:
+          # Legacy support for Redis Cluster driver. Driver dropped in v3.0.0.
+          - name: REDIGOCLUSTER_SHARDCOUNT
+            value: "128"
+          - name: TYK_PMP_ANALYTICSSTORAGECONFIG_DATABASE
+            value: "{{ .Values.redis.storage.database }}"
+          - name: TYK_PMP_ANALYTICSSTORAGECONFIG_REDISUSESSL
+            value: "{{ default "false" .Values.redis.useSSL }}"
+          - name: TYK_PMP_ANALYTICSSTORAGECONFIG_ADDRS
+            value: {{ include "tyk-hybrid.redis_url" . | quote }}
+          - name: TYK_PMP_ANALYTICSSTORAGECONFIG_ENABLECLUSTER
+            value: "{{ default "false" .Values.redis.enableCluster }}"
+          - name: TYK_PMP_PUMPS_HYBRID_TYPE
+            value: "{{ default "hybrid" .Values.pump.type }}"
```

Comment:
Why are we adding the ability to change the pump type? if someone changes that why do we add the rest of the hybrid pump config?